core/modules/taxonomy/taxonomy.module

```diff
@@ -1482,11 +1489,124 @@ function taxonomy_field_formatter_view($entity_type, $entity, $field, $instance,
         );
       }
       break;
+
+    case 'taxonomy_term_reference_rendered':
+      // To prevent infinite recursion caused by reference cycles, we store
+      // diplayed terms in a recursion queue.
+      $recursion_queue = &backdrop_static(__FUNCTION__, array());
+
+      // If no 'referencing entity' is set, we are starting a new 'reference
+      // thread' and need to reset the queue.
+      // @todo Bug: $entity->referencing_entity on terms referenced in a different
+      // thread on the page. E.g: 1 references 1+2 / 2 references 1+2 / visit homepage.
+      // We'd need a more accurate way...
```

Comment:
I don't quite follow this example. If it's possible to fix this "bug" before merging, we should attempt to do that.

Reply:
It also doesn't look like Taxonomy Term Reference fields set a "referencing_entity" anyway. So this won't work for core's built-in taxonomy term references.